d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -43,16 +43,43 @@ interface ResourceWatcher
   interface NodeResourceWatcher extends ResourceWatcher
   {
     void onChanged(NodeUpdate update);
+
+    /**
+     * onDelete method for interface functional completeness
+     * We will explicitly ignore it. We still want to call it to ensure that the callback is invoked quickly with the
+     * cached value
+     *
+     * @param resourceName
+     */
+    void onDelete(String resourceName);
```

Comment:
This is problematic. `onChanged` doesn't take in the resource name, and neither should this. This is because the property name in the EventBus is `FooCluster`, while `resourceName` here is actually `/d2/clusters/FooCluster`, so it's a mismatch. You should remove the `resourceName` parameter altogether to avoid this mistake.